orderbook/src/orderbook.rs

```diff
@@ -39,11 +49,42 @@ impl Orderbook {
     }
 
     pub async fn get_orders(&self, filter: &OrderFilter) -> Result<Vec<Order>> {
-        self.storage.get_orders(filter).await
+        let orders_without_balance = self.storage.get_orders(filter).await?;
+
+        // Since order can come from storage after a cold start there is the possibility that they are not yet registered
+        // for balance updates. In this case we do it here.
+        let untracked = orders_without_balance
+            .iter()
+            .filter_map(|order| {
+                match self
+                    .balance_fetcher
+                    .get_balance(order.order_meta_data.owner, order.order_creation.sell_token)
+                {
+                    Some(_) => None,
+                    None => Some((order.order_meta_data.owner, order.order_creation.sell_token)),
+                }
+            })
+            .collect();
+        self.balance_fetcher.register_many(untracked).await;
+
+        // Enrich orders with balance information
+        Ok(orders_without_balance
+            .into_iter()
+            .map(|mut order| {
+                order.order_meta_data.available_balance = self
+                    .balance_fetcher
+                    .get_balance(order.order_meta_data.owner, order.order_creation.sell_token);
+                order
+            })
+            .collect())
```

Comment:
I would do this in a non functional manner and update the orders in place instead of creating a new vector
```rust
for order in orders_without_balance.iter_mut() {
    order.order_meta_data.available_balance = ...
}

Reply:
In the next PR I'm using the iterator to also filter orders that don't have enough balance. Is there a syntactically nice way to do it non-functionally?

Reply:
Maybe Vec::retain